flang/lib/Lower/CallInterface.cpp

```diff
@@ -166,9 +171,27 @@ bool Fortran::lower::CalleeInterface::hasAlternateReturns() const {
 }
 
 std::string Fortran::lower::CalleeInterface::getMangledName() const {
-  return funit.isMainProgram()
-             ? fir::NameUniquer::doProgramEntry().str()
-             : converter.mangleName(funit.getSubprogramSymbol());
+
+  if (funit.isMainProgram())
+    return fir::NameUniquer::doProgramEntry().str();
+  // Do NOT mangle names for functions/subroutines with BIND(C) attribute.
+  // These could be potential calls to external RTL (i.e OpenMP), that may not
+  // follow the same mangling semantics.
+  if (Fortran::semantics::IsBindCProcedure(funit.getSubprogramSymbol())) {
+    const auto &symbol = funit.getSubprogramSymbol();
```

Comment:
Could you move the content of this block into a `static std::string getBindCName(const Symbol&);` helper in this file ?
You must also use it in `CallerInterface::getMangledName`.